### PR TITLE
Clean python api 

### PR DESCRIPTION
**Pull request by baiyfbupt**

### CODE REVIEW

**Comment by qingqing01** [comment]
```
+paddle.fluid.layers.lod_rank_table ArgSpec(args=['x', 'level'], varargs=None, keywords=None, defaults=(0,))
+paddle.fluid.layers.max_sequence_len ArgSpec(args=['rank_table'], varargs=None, keywords=None, defaults=None)
+paddle.fluid.layers.lod_tensor_to_array ArgSpec(args=['x', 'table'], varargs=None, keywords=None, defaults=None)
+paddle.fluid.layers.array_to_lod_tensor ArgSpec(args=['x', 'table'], varargs=None, keywords=None, defaults=None)
```

These APIs have been removed. But the latest `whl` still will generate these interfaces. @reyoung @typhoonzero @panyx0718

**Review by qingqing01 — APPROVED** [review]
LGTM.

**Comment by baiyfbupt** [comment]
@panyx0718 @typhoonzero This PR move some Python API to the right place, please help to review.Many thanks!

File: paddle/fluid/API.spec [hunk 1]
```diff
@@ -170,6 +170,7 @@ paddle.fluid.layers.mean_iou ArgSpec(args=['input', 'label', 'num_classes'], var
 paddle.fluid.layers.relu ArgSpec(args=['x'], varargs=None, keywords=None, defaults=None)
 paddle.fluid.layers.log ArgSpec(args=['x'], varargs=None, keywords=None, defaults=None)
 paddle.fluid.layers.crop ArgSpec(args=['x', 'shape', 'offsets', 'name'], varargs=None, keywords=None, defaults=(None, None, None))
+paddle.fluid.layers.rank_loss ArgSpec(args=['label', 'left', 'right', 'name'], varargs=None, keywords=None, defaults=(None,))
```
Comment (panyx0718) [inline paddle/fluid/API.spec:173]: where is this from?

Comment (baiyfbupt) [reply paddle/fluid/API.spec:173]: It was added by this PR https://github.com/PaddlePaddle/Paddle/pull/12132 ,but `API.spec` was not updated in time

Comment (panyx0718) [reply paddle/fluid/API.spec:173]: You can merge #12132 without updating API.spec? That seems like a bug in API.spec check?

**Review by panyx0718 — APPROVED** [review]

File: paddle/fluid/API.spec (hunk 1 above)
Comment (baiyfbupt) [reply paddle/fluid/API.spec:173]: It seems that just adding an API interface will not trigger the blocking of CI.

**Pull request merged**